lib/rules/no-side-effects.js

```diff
@@ -18,42 +19,42 @@ module.exports = {
   },
 
   create(context) {
+    let emberImportAliasName;
+
     const message = 'Don\'t introduce side-effects in computed properties';
 
     const report = function (node) {
       context.report(node, message);
     };
 
     return {
+      ImportDeclaration(node) {
+        emberImportAliasName = ember.getEmberImportAliasName(node);
```

Comment:
I don't think this is quite correct. A given module will have _many_ import statements, with the logic here we are clobbering `emberImportAliasName` for each one.

Can you add the following good test:

```js
import Ember from 'ember';
import Foo from "some-other-thing"; 
let foo = computed("test", function() { 
  Foo.set(this, "testAmount", test.length); 
  return ""; 
});
```